backend/app/api/auth.py

```
@@ -125,23 +145,29 @@ async def oauth_callback(
         server_id = await oauth_flow.complete_authorization(state, code)
     except KeyError:
+        logger.warning(
+            "OAuth callback carried an unknown or expired state — the sign-in was "
+            "superseded by a newer attempt, timed out, or the control plane restarted "
+            "mid-flow. Start the sign-in again."
+        )
         if stopped:
             sup.nudge()  # unknown state; bring the stopped bridge back with existing tokens
-        return _oauth_redirect(_ERROR_REDIRECT)
+        return _oauth_error(_REASON_EXPIRED)
     except Exception as exc:  # token exchange / provider error
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)